s3transfer/copies.py

```diff
@@ -365,4 +366,14 @@ def _main(
         for callback in callbacks:
             callback(bytes_transferred=size)
         etag = response['CopyPartResult']['ETag']
-        return {'ETag': etag, 'PartNumber': part_number}
+        part = {'ETag': etag, 'PartNumber': part_number}
+        checksums = [
+            'ChecksumCRC32',
+            'ChecksumCRC32C',
+            'ChecksumSHA1',
+            'ChecksumSHA256',
+        ]
+        for checksum in checksums:
+            if checksum in response['CopyPartResult']:
+                part[checksum] = response['CopyPartResult'][checksum]
+        return part
```

Comment:
Thanks for the feedback, it makes sense to handle checksums in a flexible way.

Unfortunately, `ChecksumAlgorithm` is not part of `extra_args` since it's only passed in `CreateMultipartUploadTask`. And if we add it to `UPLOAD_PART_COPY_ARGS` so it's part of `extra_args` it will raise `ParamValidationError` since upload part doesn't accept that.

So my intuition told me to grab it from the general `copy` `extra_args` and pass it to `CopyPartTask` as a new parameter (done in 81a4c339f8d810b6737122b8999a9f4b00402e0c), another possible solution that I see is to add it to `extra_args` and extract it (using `.pop()`) before calling `client.upload_part_copy`. If you think the latest approach is the better I can make that change.